applications/visualizer/frontend/src/components/viewers/ThreeD/STLMesh.tsx

```diff
@@ -22,10 +23,17 @@ const STLMesh: FC<Props> = ({ id, color, opacity, renderOrder, isWireframe, stl
   const { workspaces } = useGlobalContext();
   const workspaceId = useSelector((state: RootState) => state.workspaceId);
   const workspace: Workspace = workspaces[workspaceId];
+
   const onClick = (event: ThreeEvent<MouseEvent>) => {
     const clicked = getFurthestIntersectedObject(event);
+    const { id } = clicked.userData;
+    const isNeuronSelected = workspace.selectedNeurons.has(id);
     if (clicked) {
-      workspace.toggleSelectedNeuron(clicked.userData.id);
+      if (isNeuronSelected) {
+        workspace.removeSelectedNeuron(id).removeSelection(id, ViewerType.Graph);
```

Comment:
As `removeSelectedNeurons` and `addSelectedNeuron` will be removed, there call also there, it will simplify things :)